shared/src/commonMain/kotlin/com/mbta/tid/mbta_app/cache/ResponseCache.kt

```diff
@@ -2,67 +2,87 @@ package com.mbta.tid.mbta_app.cache
 
 import io.ktor.client.plugins.ResponseException
 import io.ktor.client.statement.HttpResponse
+import io.ktor.client.statement.bodyAsText
 import io.ktor.http.HttpStatusCode
+import io.ktor.http.etag
 import kotlin.time.Duration
 import kotlin.time.Duration.Companion.hours
 import kotlin.time.TimeSource
 import kotlinx.coroutines.sync.Mutex
 import kotlinx.coroutines.sync.withLock
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
 import org.koin.core.component.KoinComponent
+import org.koin.core.component.inject
+import org.koin.core.parameter.parametersOf
 
-sealed class ConditionalResponse {
-    data object NotModified : ConditionalResponse()
+@Serializable
+data class Response(
+    val etag: String?,
+    @Serializable(with = TimeMarkSerializer::class)
+    var fetchTime: TimeSource.Monotonic.ValueTimeMark,
+    val data: String
```

Comment:
This double encodes the JSON response body, which feels gross, but I don't see another good way to make this support arbitrary response types. I think ideally we'd make `ResponseCache` generic and handle all serde internally, but [Koin doesn't play well](https://insert-koin.io/docs/reference/koin-core/definitions/#dealing-with-generics) with generics. 

Since the `ResponseCache` koin module is just a factory instead of a singleton now, we might be able to get away with defining them normally, as needed in the singleton repositories, but there's still some awkwardness with Json decoding also not loving generics, so we'd still need to pass in a serializer separately.

Reply:
Yeah I considered it, but it seemed cleanest from an implementation standpoint to have them colocated, and it's harder for them to get out of sync this way. If anything, it probably makes the most sense to put it in a DataStore.

Reply:
Double-encoding 3 MB of JSON might just be worse.

Reply:
Yeah fair, that'll also mean we don't need to rewrite the entire 3MB file every time notmodified updates the timestamp. Cool, I'll start on moving them into DataStore.

Reply:
Making the `ResponseCache` generic seems like it'd be worth doing regardless. Unfortunately, you can't have an `inline constructor` to automatically determine the serializer, but you can declare an `inline fun` on the `companion object` to get a similar effect: https://pl.kotl.in/fH4yXhJaq

Reply:
Oh interesting, yeah that's clean, thanks for the suggestion!

Reply:
Okay, made both of those changes. I'm much more satisfied with this! I ended up just tossing the metadata into another json file after all, DataStore seemed like more trouble than it was worth.